Dev 2021

lbednarczyk opened the pull request.

lbednarczyk committed:
Updated explore page

lbednarczyk committed:
Updated what is ancova section of prerequisites

lbednarczyk committed:
Updated prerequisite page

lbednarczyk committed:
Formatting and commented out unnecessary code

lbednarczyk committed:
Added description file

lbednarczyk committed:
Moved files to docs folder

lbednarczyk committed:
Combined ui and server

lbednarczyk committed:
Renamed server and ui

lbednarczyk committed:
Renamed server and ui

lbednarczyk committed:
Added docs folder

lbednarczyk committed:
Fixed picture rendering

lbednarczyk committed:
Fixed error

lbednarczyk committed:
Changed game type

lbednarczyk committed:
Added plot images to prereqs

lbednarczyk committed:
Changed arrows in prereqs

lbednarczyk committed:
Deleted unused code

lbednarczyk committed:
Deleted ui and server files

lbednarczyk committed:
Deleted unused packages

lbednarczyk committed:
Added alt text

lbednarczyk committed:
Added screenshot file

shravanisamala commented on app.R:
```diff
@@ -0,0 +1,1123 @@
+                         interpreting interaction plots.'),
+                br(),
+                h2('Instructions:'),
+                p(
```
For lines 90 - 116, instead of doing p() and the tags$li() try: 
tags$ul(
     tags$li(""), 
     tags$li(""), 
     tags$li(""), 
)
This way the bullet points won't look like they are so are so far to the left.

shravanisamala commented on app.R:
```diff
@@ -0,0 +1,1123 @@
+                    bsButton(inputId = "go",
+                             label = "Prerequisites",
+                             icon = icon("book"),
+                             style = "danger",
```
Think this should be "default" instead of "danger"

shravanisamala commented on app.R:
```diff
@@ -0,0 +1,1123 @@
+                       "Example: Who makes the most money? Will gender or 
+                             years after graduation influence the income?")
+                  )
+                ),
```
Maybe add a break here just so it looks a little spaced out

shravanisamala commented on app.R:
```diff
@@ -0,0 +1,1123 @@
+                      bsButton(inputId = "start",
+                               label = "Explore!",
+                               icon = icon("bolt"),
+                               style = "danger",
```
Change "danger" to "default"

shravanisamala commented on app.R:
```diff
@@ -0,0 +1,1123 @@
+                  mainPanel(
+                    plotOutput('plot_gg'),
+                    tags$b(verbatimTextOutput('analysis1')),
+                    div(style = "text-align: center",
```
Because the button is in the main panel, it does not show up centered on the page. I think you can try removing it from the main panel and just have in the page itself so it is centered!

shravanisamala commented on app.R:
```diff
@@ -0,0 +1,1123 @@
+                  )
+                )
+                ),
+        tabItem(tabName='game',
```
When I tried opening the game page it did not work because it said no package called sortable was there. I think that the sortable package needs to be at the top with all of the packages because I had not downloaded that package before but once I did and put it at the top, the app worked.

shravanisamala commented on app.R:
```diff
@@ -0,0 +1,1123 @@
+  )
+)
+
+#Server ----
```
Is this section where you are reading in data supposed to be in the server?

shravanisamala commented on app.R:
```diff
@@ -0,0 +1,1123 @@
+                     dist = "nonrandom", 
+                     formula = a, 
+                     id = "id")
+      def <- defData(def,
```
is def the inputId? If so, I think labeling it would be helpful!

shravanisamala reviewed (changes requested):
Good job! I just left a few comments

shravanisamala commented on app.R:
```diff
@@ -0,0 +1,1123 @@
+                p("This app was developed and coded by Luxin Wang and
+                         modified by Zhiruo Wang and Lydia Bednarczyk."),
+                uiOutput('ack2')
+        ),
```
Don't forget the updated by statement

shravanisamala reviewed (changes requested):
I just added one new comment about the updated by statement

lbednarczyk committed:
Formatting updates

adpoleski1 reviewed (dismissed):
Great work!  Everything looks good to me

shravanisamala commented on app.R:
```diff
@@ -0,0 +1,1118 @@
+library(shiny)
+#library(png)
```
You can probably remove the packages that aren't being used!

shravanisamala commented on app.R:
```diff
@@ -0,0 +1,1118 @@
+                br(),
+                h2('Acknowledgements:'),
+                p("This app was developed and coded by Luxin Wang and
+                         modified by Zhiruo Wang and Lydia Bednarczyk."),
```
Maybe add the year you updated in because if someone else updates it later the update statement below will have their name and year

shravanisamala reviewed (dismissed):
Everything works great! I just made a suggestion about the acknowledgements part so good job :)

lbednarczyk committed:
Updated references.

lbednarczyk committed:
Added link to ANOVA assumptions app.

lbednarczyk committed:
Updates

neilhatfield committed:
Partial Update

This is a partial update

Phichchaya27 reviewed (dismissed):
“Recommended Priority: Urgent”
1) On the explore page, the "Download the Data" can be deleted
2) On the explore page, the plots that show up should have units in the labels

Phichchaya27 committed:
Edit code in app.R

feedback in the game page

Phichchaya27 committed:
Update app.R

Delete the download data button

Phichchaya27 committed:
Edit references

Phichchaya27 committed:
edit reference

Phichchaya27 committed:
edit reference

Phichchaya27 committed:
edit code in explore page

Add units in plot labels

Phichchaya27 committed:
edit the reference page

delete unused references

Phichchaya27 committed:
Edit app.R in reference page

add references for datasets

Phichchaya27 committed:
Delete README.html file

Phichchaya27 committed:
Update DESCRIPTION

change date and add name

JingFu1 reviewed (changes requested):
Recommended Priority: Urgent

According to the style guide, I think "ctb" is better than "aut"

Phichchaya27 committed:
Update DESCRIPTION

Phichchaya27 committed:
edit acknowledgement

Phichchaya27 committed:
Update app.R

update last update

sv101 reviewed (dismissed):
Looks good!

neilhatfield reviewed (changes requested):
I have a few changes I would like you to make and some things to investigate. On the whole the app appears to be in pretty decent shape.
### DESCRIPTION
- [x] Change the title to Analysis of Covariance
- [x] Change the date to 2018-08-01
- [x] Change the Lifecycle to questioning
- [x] Change the Learning Outcome to "The student will learn to use interaction plots to decide whether interaction terms are significant in ANCOVA settings."

### README.md
- [x]  Spell out ANCOVA in the title
- [x] Add "# App Description" above the description that is there.

### Files
- [x] Delete model2.csv as this does not appear to be used.
- [x] Delete all image files from the `www` folder which are no longer used. For example, arrow.gif, arrow.png, check.png, etc.
- [x] Delete the navcolor.css file from the `www` folder

### app.R
- [x] Delete `library(fontawesome)` call and update the reference section accordingly.
- [x] Uncomment out `library(sortable)` as this package is needed
- [x] Add a section header for the game page in the UI definition (i.e., "### Game page ----")
- [x] Move the code that loads the data sets from between the UI and server to the labelled section just above the UI
- [x] Fix the formatting of the `sendSweetAlert` for the info button (remove the colon and add `type = "info"`)
- [x] Change all instances of snake_case to camelCase to be aligned with the Style Guide
- [x] Check spelling and grammar. (E.g., "continous" should be "continuous"; "Select covariance" should be "Select covariate")
- [x] Change any `tags$b` to `tags$strong`
- [x] Put one space on both sides of symbols such as `=`, `==`, and `<-`

### Things to Investigate and Report Back On
- [ ] I don't understand the presence of the Diagnostic Plots subsection of the Prereq's. I didn't see where those diagnostic plots are getting used elsewhere in the app. Did I miss something? If such plots aren't being used, then we should remove them from the prereq's
- [ ] Currently, the app displays raw R output for the anova table. Should we convert this to a more polished version (like what I require in class)?
- [ ] Why does the app fit models for the otter and diet data sets and then use predicted data for making the plots? Why aren't we just using the actual data for the plots?
- [ ] Where does the `output$p` go? I can't find a corresponding UI part. Is this something that we need to add to the UI somewhere?
- [ ] What is the role of `maxScore` and the commented out code towards the end of the server?

Phichchaya27 committed:
change DESCRIPTION, README, app.R, FILES

Phichchaya27 committed:
Update app.R

neilhatfield committed:
Update README.md

luqije reviewed (changes requested):
Report my reviews for "Things to Investigate and Report Bacnk On":
- The diagnostic plots are not necessarily needed in Pre. Because later in the explore page and game page, we are not showing information related to these. Also, the app "Assumptions of ANOVA" offers information related to ANCOVA. 
- We should convert the R output to a polished version. Since not every user uses R when analyzing ANCOVA. Displaying the output using the ds version (used in the paper) will be better.
- I think using a random dataset on Explore page can help the users more directly to see how things changes can affect the interaction with the p-value through the plot. However, in this case, we can add information about the meaning behind them. By practicing this, the users can explore the matching game on the Game page.
- `output$p` is missing in the UI part of the Explore page. I believe the author is trying to show the feedback of what is the meaning of the p-value of each output with different datasets.
- `maxScore` is missing in the UI part of the Game page. I think it's meant for calculating the total score of each trial. 

Error and Suggestions:
- The plots for the Diet dataset when choosing the variable Age with Gender and Pre-diet Weight with Gender are not showing with errors. 
- Add brake between each explanation in Pre would be helpful.

Sburke26 reviewed (changes requested):
Here is what I noticed while reviewing:

There appears to be two README files.
The year badge in the README isn't updated properly.
The info pop-up does not have a circled i icon.
The app does not run with the object named "abChange". I had to change it to "ab_change" for it to run.
The prerequisites page could be improved to be more attractive and beneficial to the user.
In particular, removing the diagnostic plots is one way, as it appears they don't support what is covered in the app. Also, maybe switching to a collapsible tab for each type of analysis instead of a spaced out table would be better. When the window of the app is minimized, this spaced out table is hard to look at.
There also appears to be a few instances of commented out code. Some of which may come in handy if the app were to try to implement certain features (ex.score,).
There are also some issues with spacing in the code.

Edit: 
Also I noticed there was a console message: "Warning in is.na(x) : is.na() applied to non-(list or vector) of type 'closure'".

Commit:
update app.R

have trouble generating three different plot with different models.

Commit:
update for loop issue

neilhatfield committed:
Update app.R

Removed some old code related to downloading data.
Used lapply to generate plots and tables on matching game page.

Commit:
update

rank_list, submit, and newChallenge are not working

neilhatfield committed:
Getting Sortable to work

Here is the updated code for moving the plots and tables into sortables.

Commit:
update R

updated the models, submit button is working, newChallenge button is working with minor thing needs to be fixed
